src/System.Runtime.CompilerServices.Unsafe/ref/System.Runtime.CompilerServices.Unsafe.cs

```diff
@@ -17,7 +17,11 @@ public static partial class Unsafe
         public static unsafe void* AsPointer<T>(ref T value) { throw null; }
         public static unsafe ref T AsRef<T>(void* source) { throw null; }
         public static ref T AsRef<T>(in T source) { throw null; }
-        public static T As<T>(object o) where T : class { throw null; }
+#if NETSTANDARD2_1
+        [return: System.Diagnostics.CodeAnalysis.MaybeNull]
+        [return: System.Diagnostics.CodeAnalysis.NotNullIfNotNull("o")]
+#endif
+        public static T As<T>(object? o) where T : class? { throw null; }
```

Comment:
Why not this?

```csharp
public static T? As<T>(object? o) where T : class { throw null; }
```

Reply:
Is that better in some way?  With the existing one I could use `As<string?>` if I wanted to.

Reply:
Was just wondering why this form was chosen 😄

Reply:
Yeah. :smile: I used the `MaybeNull` in place of the `T?`.